### PR TITLE
Mesh 1323: use of scheduler pallet in bridge

### PR DESCRIPTION
**Pull request by vkomenda**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -36,9 +36,9 @@ dependencies = [
 
 [[package]]
 name = "aes"
-version = "0.5.0"
+version = "0.4.0"
```
Comment (Centril) [inline Cargo.lock:39]: Hmm; why is everything being downgraded in the lockfile?

File: pallets/bridge/src/lib.rs [hunk 2]
```diff
@@ -393,16 +394,47 @@ decl_event! {
...
-            Self::handle_timelocked_txs(block_number)
+        fn on_runtime_upgrade() -> Weight {
+            use frame_support::{migration::StorageKeyIterator, Twox64Concat};
+
+            let now = frame_system::Module::<T>::block_number();
+
+            // Migrate timelocked transactions.
+            StorageKeyIterator::<T::BlockNumber, Vec::<BridgeTx<T::AccountId, T::Balance>>, Twox64Concat>::new(b"Bridge", b"TimelockedTxs")
+                .drain()
+                .filter_map(|(block_number, txs)| {
```
Comment (Centril) [inline pallets/bridge/src/lib.rs:410]: This seems to be infallible, so `.map(...)` is sufficient?

File: pallets/bridge/src/lib.rs [hunk 3]
```diff
@@ -393,16 +394,47 @@ decl_event! {
...
+                    // Schedule only for future blocks.
+                    let block_number = T::BlockNumber::max(block_number, now + One::one());
+                    let calls: Vec<_> = txs
+                        .into_iter()
+                        .map(|tx| {
+                            Call::<T>::handle_scheduled_bridge_tx(tx).into()
+                        })
+                        .collect();
```
Comment (Centril) [inline pallets/bridge/src/lib.rs:418]: Although this is a one-time-code thing, consider a `.flat_map(...)` instead for efficiency (don't allocate into a `Vec<_>` that is consumed immediately).

File: pallets/bridge/src/lib.rs [hunk 4]
```diff
@@ -688,45 +733,32 @@ impl<T: Trait> Module<T> {
...
+        // Schedule the transaction as a dispatchable call.
+        let call = Call::<T>::handle_scheduled_bridge_tx(bridge_tx.clone()).into();
+        // FIXME: handle errors.
+        if let Err(e) = T::Scheduler::schedule(
+            DispatchTime::At(unlock_block_number),
+            None,
+            LOWEST_PRIORITY,
+            RawOrigin::Root.into(),
+            call,
+        ) {
```
Comment (Centril) [inline pallets/bridge/src/lib.rs:750]: Call seems substantially similar to the one above; perhaps refactor?

**Review by Centril — COMMENTED** [review]
Haven't seen this code before, but these things stood out to me.

File: Cargo.lock (hunk 1 above)
Comment (vkomenda) [reply Cargo.lock:39]: Hmm, possibly a cargo-remote artefact. Not sure.

**Review by maxsam4 — APPROVED** [review]

File: pallets/bridge/src/lib.rs (hunk 3 above)
Comment (vkomenda) [reply pallets/bridge/src/lib.rs]: I've changed that to a `.map` without collecting into a `Vec`. Please check if that's OK with you.

**Review by Centril — APPROVED** [review]

**Review by satyamakgec — APPROVED** [review]
LGTM

**Pull request merged**